src/prefect/cli/deploy.py

```diff
@@ -1326,6 +1331,30 @@ def _check_for_matching_deployment_name_and_entrypoint_in_prefect_file(
     return False
 
 
+def _check_if_identical_deployment_in_prefect_file(
+    untemplated_deploy_config: Dict,
+) -> bool:
+    """
+    Check if the given deploy config is identical to an existing deploy config in the
+    prefect.yaml file, meaning that there have been no updates and prompting to save is unnecessary.
+    """
+
+    templated_deploy_config = _format_deployment_for_saving_to_prefect_file(
+        untemplated_deploy_config
+    )
+
+    prefect_file = Path("prefect.yaml")
+    if prefect_file.exists():
+        with prefect_file.open(mode="r") as f:
+            parsed_prefect_file_contents = yaml.safe_load(f)
+            deployments = parsed_prefect_file_contents.get("deployments")
+            if deployments is not None:
+                for _, existing_deployment in enumerate(deployments):
+                    if existing_deployment == templated_deploy_config:
```

Comment:
That's a good question! The `templated_deploy_config` would be taken straight from the `prefect.yaml` when they are using an existing configuration, so if there aren't changes to it it would be identical. If there are changes to it then we would prompt to save.

If for some reason they happened to specify a configuration with the same `name` and `entrypoint` and nothing else was changed, these dictionaries would still be equal as they'll still have the same k/v regardless of how the user orders them in their `prefect.yaml`. They will not be prompted to save with a deployment with the same entrypoint and name.

Reply:
> The templated_deploy_config would be taken straight from the prefect.yaml

This is the piece that I was missing!

Reply:
I have updated the variables to be more clear, thanks!